doc/media-settings/Media-based-Port-settings.md

```diff
@@ -0,0 +1,281 @@
+
+                    "lane3":"0x443322"
+
+                },
+
+                "idriver": {
+
+                    "lane0":"0xb",
+
+                    "lane1":"0xc",
+
+                    "lane2":"0xd",
+
+                    "lane3":"0xa"
+
+                }
+
+            },
+
+            "DELL-5678": {
+
+                "preemphasis": {
+
+                    "lane0":"0x102233",
+
+                    "lane1":"0x132233",
+
+                    "lane2":"0x152244",
+
+                    "lane3":"0x413322"
+
+                },
+
+                "idriver": {
+
+                    "lane0":"0xc",
+
+                    "lane1":"0xc",
+
+                    "lane2":"0xd",
+
+                    "lane3":"0xb"
+
+                }
+
+            }
+
+        }
+
+    }
+
+}
+
+![](https://github.com/dgsudharsan/SONiC/blob/hld_media/doc/media-settings/event_flow.png)
+
+Flow:
+=====
+
+              When a media is detected in xcvrd daemon, it constructs the key identifier string as discussed above and searches the media_settings.json file. If found the key value pairs are fetched. The xcvrd daemon notifies these key value pairs along with the alias port to portsorch. The portsorch task will convert the alias to SAI object Id and notifies the syncd by doing port attribute set of corresponding settings. Syncd invokes the SAI port set those attribute with specified values. The SAI implementation then programs the attribute into the hardware.
+
+              The notification of hardware profile from xcvrd to portsorch task will be done during initialization and during media detect event. This is not required during media removal event. Since the media settings are required only for the proper functioning of the optics or DAC, the handling can be restricted to media insert event alone and no action needs to be taken during media removal.
+
+              This mechanism is also very helpful in supporting new media types without upgrading the Operating system. If a new media type need to be supported the only change that needs to be done is modify media_settings.json to add the new media type.
+              
+![](https://github.com/dgsudharsan/SONiC/blob/hld_media/doc/media-settings/key_selection_flow.png)
+
+Breakout Scenario:
+==================
+
+              For breakout the particular media, if it has a global value is listed in the media_settings.json with appropriate lane values. If the values are specific to a port, then the entry is listed under the port. If the logical port is going to be created after breakout (E.g Ethernet0 is modified to Ethernet0,Ethernet1,Ethernet2,Ethernet3) those corresponding ports are listed in the ports section in media_settings.json. In the current breakout scenario, when port_config.ini is modified and a config reload is done, the flow will be similar to bootup sequence and thus all the media settings will be pushed from xcvrd to portsorch on restart during config apply.
```

Comment:
can you give detailed example. have difficult to understand.

Reply:
Hi @lguohan I have given an example of breakout. I have modified the design of media settings file slightly so that the users do not need to do anything additionally to handle breakout scenario